prow/cmd/deck/main.go

```diff
@@ -298,7 +299,9 @@ func main() {
 	health.ServeReady()
 
 	// setup done, actually start the server
-	logrus.WithError(http.ListenAndServe(":8080", traceHandler(mux))).Fatal("ListenAndServe returned.")
+	server := &http.Server{Addr: ":8080", Handler: traceHandler(mux)}
+	interrupts.RunOrShutdown(server, 5*time.Second, logrus.StandardLogger())
+	logrus.Info("Deck is shutting down...")
```

Comment:
nit: I think these log messages are redundant with the one emitted by `RunOrShutdown`. They are also a bit misleading because read as if shut down logic is starting when in reality it is complete when they are reached.